moveit_kinematics/CMakeLists.txt

```diff
@@ -32,12 +32,12 @@ set(THIS_PACKAGE_INCLUDE_DIRS
 catkin_package(
   LIBRARIES
   INCLUDE_DIRS
-    ${EIGEN3_INCLUDE_DIRS}
     ${THIS_PACKAGE_INCLUDE_DIRS}
   CATKIN_DEPENDS
     pluginlib
     moveit_core
     moveit_ros_planning
+    DEPENDS Eigen3
```

Comment:
Passing on Eigen dependencies in ROS is a bit of a curiosity really. It was/(is?) broken in many places and nobody ever notices, because every dependent project still includes it *somehow* either directly or through different dependency.
It doesn't surprise me that travis succeeds even when your fix probably does not work.
This is only made worse by the fact that until a couple of months ago only Debian's version of eigen3 provided the variable `EIGEN3_INCLUDE_DIRS` (notice the trailing `S`) and everyone else only set `EIGEN3_INCLUDE_DIR` (no `S`). Anyway..

> But , I didn't get it.

Variables in cmake are case-sensitive.
`catkin_package`'s `DEPENDS <package>` field invokes some logic that automatically adds the contents of the variables `<package>_INCLUDE_DIRS` and `<package>_LIBRARIES` to projects that find_package this catkin package. See [here](https://github.com/ros/catkin/blob/kinetic-devel/cmake/catkin_package.cmake#L168).

So with the code you currently propose `catkin_package` will look at the variables `Eigen3_INCLUDE_DIRS` and `Eigen3_LIBRARIES`.
eigen is header-only, so the latter doesn't matter.
But the variable `Eigen3_INCLUDE_DIRS` doesn't exist either. It should be looking for `EIGEN3_INCLUDE_DIRS` instead.

The newline is just a matter of taste and imho fits better with the rest of the CMakeLists.txt.